scaffolder-templates/basic-workflow/template.yaml

```diff
@@ -335,6 +335,19 @@ spec:
         repoUrl: github.com?owner=${{ parameters.orgName }}&repo=${{ parameters.repoName }}-gitops
         defaultBranch: main
         sourcePath: gitops
+
+    # TODO find out how to skip this if this if the argocd scaffolder plugin isn't installed
+    - id: create-argocd-resources
+      if: ${{ parameters.CI == 'tekton_argocd' }}
+      name: Create ArgoCD Resources
+      action: argocd:create-resources
+      input:
+        appName: ${{ parameters.component_id }}-bootstrap
```

Comment:
There needs to be two applications, one for each component:
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: yfassess
  namespace: orchestrator-gitops
  labels:
    app-type: bootstrap
    app: yfassess
spec:
  destination:
    namespace: sonataflow-infra
    server: https://kubernetes.default.svc
  project: orchestrator-gitops
  source:
    path: argocd
    repoURL: git@github.com:test-workflows/yfassess.git
    targetRevision: HEAD
  syncPolicy:
    automated:
      # Keep history of pipeline runs
      prune: false
      selfHeal: true
```
and
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: yfassess-gitops
  namespace: orchestrator-gitops
  labels:
    app-type: workflow
    app: yfassess
spec:
  destination:
    namespace: sonataflow-infra
    server: https://kubernetes.default.svc
  project: orchestrator-gitops
  source:
    path: ./kustomize/base
    repoURL: git@github.com:test-workflows/yfassess-gitops.git
    targetRevision: HEAD
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
```

Reply:
> There needs to be two applications, one for each component:
> 
> ```
> apiVersion: argoproj.io/v1alpha1
> kind: Application
> metadata:
>   name: yfassess
>   namespace: orchestrator-gitops
>   labels:
>     app-type: bootstrap
>     app: yfassess
> spec:
>   destination:
>     namespace: sonataflow-infra
>     server: https://kubernetes.default.svc
>   project: orchestrator-gitops
>   source:
>     path: argocd
>     repoURL: git@github.com:test-workflows/yfassess.git
>     targetRevision: HEAD
>   syncPolicy:
>     automated:
>       # Keep history of pipeline runs
>       prune: false
>       selfHeal: true
> ```
> 
> and
> 
> ```
> apiVersion: argoproj.io/v1alpha1
> kind: Application
> metadata:
>   name: yfassess-gitops
>   namespace: orchestrator-gitops
>   labels:
>     app-type: workflow
>     app: yfassess
> spec:
>   destination:
>     namespace: sonataflow-infra
>     server: https://kubernetes.default.svc
>   project: orchestrator-gitops
>   source:
>     path: ./kustomize/base
>     repoURL: git@github.com:test-workflows/yfassess-gitops.git
>     targetRevision: HEAD
>   syncPolicy:
>     automated:
>       prune: true
>       selfHeal: true
> ```

making an argo application from the code repo smell like a hack, specially when tekton is for that.
can you explain that workaround and what made the tekton unusable in that case?}

Reply:
@dmartinol pls correct me if I'm mistaken here.
We had a problem with creating the workflow's pipeline from a software template when the deploying cluster is behind VPN. See https://github.com/parodos-dev/workflow-software-templates/issues/24

Instead, a tekton [pipeline template](https://github.com/parodos-dev/workflow-software-templates/blob/main/scaffolder-templates/build/argocd/pipelinerun.template) is added to the source workflow repository by [the software template](https://github.com/parodos-dev/workflow-software-templates/blob/38b2cf22bd3f4be8e407d3f716c5526a485abaff/scaffolder-templates/basic-workflow/template.yaml#L238) 

That pipeline template is used by the [github action](https://github.com/parodos-dev/workflow-software-templates/blob/38b2cf22bd3f4be8e407d3f716c5526a485abaff/scaffolder-templates/build/.github/workflows/update_pipelinerun.yaml) to create a new pipeline run with the submitted commit, which triggers the build due to the argocd application that watches for changes for this source repository.

To summarize, the current process for the workflow repository:
1. Create a PR
2. Merge a PR 
3. Github action creates and commit a tekton pipeline run
4. ArgoCD capture the new pipeline run and applies it on the cluster
5. The pipeline run builds the workflow image, generate resources, publish to the gitops repo...

We can explore alternatives to this process.

Reply:
@masayag Probably we need step 0: after creating the software template, manually deploy the bootstrap application from the config repo, as explained in https://github.com/parodos-dev/onboarding-gitops/blob/main/bootstrap/README.md

> making an argo application from the code repo smell like a hack, specially when tekton is for that.
can you explain that workaround and what made the tekton unusable in that case?}

@rgolangh tekton is unusable until someone creates a PipelineRun instance to run the required tasks. IIRC, we are now using the pull update model because we tried the push model with webhooks from GH to create it automatically, but this was discarded because of the VPN networking issues.